### PR TITLE
[FEATURE] move MzTabExport to lib and extend export

### PR DESCRIPTION
**Pull request by timosachsenberg**
pulling out current MzTabExport from lfq branch seems to be a bit more involved. @cbielow this is probably most likely what you want to work with. It contains much more information and the output has been validated by PRIDE (at least for consensusXML with IDs and inference information).
I get some failing tests on my side potentially related to moving to unordered_set for the inference code ...

### CODE REVIEW

File: src/openms/include/OpenMS/METADATA/ProteinHit.h [hunk 1]
```diff
@@ -171,6 +196,11 @@ namespace OpenMS
     /// sets the coverage (in percent) of the protein hit based upon matched peptides
     void setCoverage(const double coverage);
 
+    /// returns the set of modified protein positions
+    const std::set<std::pair<Size, ResidueModification> >& getModifications() const;
+
+    /// sets the set of modified protein positions
+    void setModifications(std::set<std::pair<Size, ResidueModification> >& mods);
```
Comment (hroest) [inline src/openms/include/OpenMS/METADATA/ProteinHit.h]: can you add Python bindings for these?

File: src/openms/include/OpenMS/METADATA/ProteinIdentification.h [hunk 2]
```diff
@@ -95,6 +108,100 @@ namespace OpenMS
...
+    private:
+      /// Float data arrays
+      FloatDataArrays float_data_arrays_;
+
+      /// String data arrays
+      StringDataArrays string_data_arrays_;
+
+      /// Integer data arrays
+      IntegerDataArrays integer_data_arrays_;
     };
```
Comment (hroest) [inline src/openms/include/OpenMS/METADATA/ProteinIdentification.h]: can you add Python bindings for these?

**Comment by hroest** [comment]
looks good, can you add Python bindings the new library functions?

**Comment by timosachsenberg** [comment]
Quick question because one change is a bit controversial.
To transport quantification at protein group level (=single protein or true group) to mzTab, @jpfeuffer and me currently annotate the quantities in a float data array attached to the protein group.
While this works great some may not like that because:
- some may argue that quant. data is in the id data structure
- float data arrays are not as easy to encode in @hendrikweisser  new ID data structure

Among other alternatives, we thought about (e.g., encoding in a consensus feature) did not play well together with the final result file that contains fraction wise consensus features that map to protein ids (and not groups).

Any thoughts about that?

**Comment by hendrikweisser** [comment]
Without having been part of the previous discussions/deliberations, some thoughts from my side:
- It would be desirable if we could find a solution that is easily adaptable to the new ID data structures.
- Those datastructures can absolutely be amended to support new requirements. If we decide the best solution is to store (some) quant. data directly in there, I'm sure that wouldn't be very difficult to add.
- I'm sceptical whether "hiding" fundamental information, like quantitative data, in meta data structures is a good idea.
- If we don't have a suitable data structure for this type of quantitative data at the moment, maybe we should make one.
- A quantitative data structure could easily reference relevant parts of the new ID data structure (peptides, proteins, protein groups...). A possible design for mzTab, where we want to export ID data with optional quant. data, would be: `export(IdentificationData id_data, QuantData q_data = QuantData())`, where `q_data` contains references into `id_data` and has an empty default.

**Comment by hroest** [comment]
> If we don't have a suitable data structure for this type of quantitative data at the moment, maybe we should make one.

I agree, it should be explicit and not implicit if possible.

**Comment by cbielow** [comment]
this seems all very reasonable. For a current project it would be very neat to have the current functionality as provided by this PR in the library.
I suggest a two step process where we: address the immediate comments (python bindings) and go forward with a merge. In a second step (dev meeting?) we can come up with a solution to the data structure problem.

**Comment by lars20070** [comment]
:+1: for two small steps.

Would be good to have this PR merged before the developer meeting. I would like to see if the problem in  https://github.com/OpenMS/OpenMS/pull/3973 persists after a rebase.

File: src/openms/include/OpenMS/METADATA/ProteinHit.h (hunk 1 above)
Comment (timosachsenberg) [reply src/openms/include/OpenMS/METADATA/ProteinHit.h]: so I need an extra struct again for the set I guess

**Comment by timosachsenberg** [comment]
Ok that's fine with me to merge it before the developer meeting if that increases our chance to make some real progress on that.

**Pull request merged**